wrappers/react/src/hotTableClass.tsx

```diff
@@ -486,24 +296,18 @@ class HotTableClass extends React.Component<HotTableProps, {}> {
    * Render the component.
    */
   render(): React.ReactElement {
-    const isHotColumn = (childNode: any) => childNode.type === HotColumn;
+    const isHotColumn = (childNode: any): childNode is ReactElement => childNode.type === HotColumn;
     const children = React.Children.toArray(this.props.children);
 
     // clone the HotColumn nodes and extend them with the callbacks
     const hotColumnClones = children
-      .filter((childNode: any) => isHotColumn(childNode))
-      .map((childNode: React.ReactElement, columnIndex: number) => {
+      .filter(isHotColumn)
+      .map((childNode, columnIndex) => {
         return React.cloneElement(childNode, {
```

Comment:
Removing cloneElement is not part of this task right? I could see us removing the need by having a column context, that a map creates and sets.

Reply:
Good idea! Let me do this, thx 😊

Reply:
Thinking deeper at the code now. If we have a `Map<column index, column settings>` in the context, the column would need to know its index to read the proper entry anyway (is this what you meant by a map?).

The only way to pass column index to `HotColumn` without requiring an explicit prop AFAIK is via `cloneElement`. 

Alternatively we could be doing a separate column context per each column:

```
const hotColumns = children
      .filter(isHotColumn)
      .map((childNode, columnIndex) => (
        <HotColumnContext.Provider value={columnIndex}>
          {childNode}
        </HotColumnContext.Provider>
      ));
```

But isn't it too much? Do you think it's better than `cloneElement`? 🤔 For sure it's more idiomatic, but is it worth it?

Reply:
Awesome 😄